fastavro/reader.py

```diff
@@ -134,6 +142,87 @@ def read_boolean(fo, writer_schema=None, reader_schema=None):
     return unpack('B', fo.read(1))[0] != 0
 
 
+def read_timestamp_millis(data, writer_schema=None, reader_schema=None):
+    return datetime.datetime.fromtimestamp(data / 1000).replace(
+        microsecond=data % 1000)
+
+
+def read_timestamp_micros(data, writer_schema=None, reader_schema=None):
+    return datetime.datetime.fromtimestamp(data/1000000).replace(
+        microsecond=data % 1000000)
+
+
+def read_date(data, writer_schema=None, reader_schema=None):
+    return datetime.date.fromordinal(data)
+
+
+def read_bytes_decimal(data, writer_schema=None, reader_schema=None):
+    scale = writer_schema['scale']
+    precision = writer_schema['precision']
+
+    size = len(data)
+
+    if PY3:
```

Comment:
Instead of having `if` on every function call, do this at definition time (outside of this function)

```python
if PY3:
    def read_decimal_from_fixed(...
else:
    def read_decimal_from_fixed(...
```